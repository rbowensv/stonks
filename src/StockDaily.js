import React from 'react'

export const StockDaily = (props) => {

    const {daily} = props;

    return <div className='stock-daily'>Daily: 
            {daily ? <div>
                <div className='stock-daily--section'>
                    <div>Open: ${daily['1. open']}</div>
                    <div>Close: ${daily['4. close']}</div>
                </div>
                <div className='stock-daily--section'>
                    <div>High: ${daily['2. high']}</div>
                    <div>Low: ${daily['3. low']}</div>
                </div>
                <div>Volume: {daily['5. volume']}</div>
            </div> : <div>Something went wrong</div>}
        </div>
}