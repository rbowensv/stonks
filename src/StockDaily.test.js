import renderer from 'react-test-renderer'
import { StockDaily } from './StockDaily'

const mockDaily = {'1. open': "23.4200", '2. high': "23.4200", '3. low': "22.9200", '4. close': "22.9300", '5. volume': "201179"}

test('renders as expected', () => {
    const component = renderer.create(<StockDaily daily={mockDaily}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });