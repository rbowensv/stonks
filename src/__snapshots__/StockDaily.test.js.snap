// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="stock-daily"
>
  Daily:
  <div>
    <div
      className="stock-daily--section"
    >
      <div>
        Open: $
        23.4200
      </div>
      <div>
        Close: $
        22.9300
      </div>
    </div>
    <div
      className="stock-daily--section"
    >
      <div>
        High: $
        23.4200
      </div>
      <div>
        Low: $
        22.9200
      </div>
    </div>
    <div>
      Volume: 
      201179
    </div>
  </div>
</div>
`;
